### ISSUE TITLE
Cache JIRA issues on filesystem

### ISSUE THREAD

**Issue by bethesque** [tomasbjerre/git-changelog-lib#67]
Thanks for your work on this library. It looks like it will be a great help to us.

I have a question about the expected usage. In some of my OSS projects, I use a changelog generation tool that prepends the release notes of the current changeset to the beginning of the CHANGELOG.md file each time I do a release. 

git-changelog-lib seems to generate the entire changelog from scratch each time. When the Jira feature is enabled, this could be a seriously long running task! The only alternative I can see is to generate a temporary changelog file, and then prepend it to the existing CHANGELOG.md.

What is the recommended approach?

**Comment by tomasbjerre** [comment]
Yes it will generate it from scratch each time. And yes prepending may work for you.

Im labeling this as enhencement to investigate if this can be implemented.

**Comment by bethesque** [comment]
As a potential source of inspiration, the conventional-changelog ruby gem that I use populates the fromRef from the most recent heading (tag) in the existing CHANGELOG.md file. This has worked very well for the projects that I've used it on.

**Comment by tomasbjerre** [comment]
The GitHub issues are cached, on filesystem. If caching was implemented on filesystem also for Jira, perhaps that would solve your problem?

**Comment by bethesque** [comment]
This runs on Jenkins, with a clean environment each time, so that wouldn't help us unfortunately.
